### ISSUE TITLE
auto killed

### ISSUE THREAD

**Issue by hamnaz** [JeanLucPons/BSGS#2]
chawla@i3-6100:~/jlp/BSGS$ ./bsgs in.txt
BSGS v1.0
BabyStep:0x40000000 (2^30.00)
Start:49DCCFD96DC5DF56487436F5A1B18C4F5D34F65DDB48CB5E0000000000000000
Stop :49DCCFD96DC5DF56487436F5A1B18C4F5D34F65DDB48CB5EFFFFFFFFFFFFFFFF
Keys :16
Number of CPU thread: 4
BabyStep Thread 0: 0x1 -> 0x10000000
BabyStep Thread 1: 0x10000001 -> 0x20000000
BabyStep Thread 2: 0x20000001 -> 0x30000000
BabyStep Thread 3: 0x30000001 -> 0x40000000
[3.02 MKey/s][Cnt 2^28.48][01:55][3632.1MB]  Killed
chawla@i3-6100:~/jlp/BSGS$ ./bsgs in.txt
BSGS v1.0
BabyStep:0x40000000 (2^30.00)
Start:49DCCFD96DC5DF56487436F5A1B18C4F5D34F65DDB48CB5E0000000000000000
Stop :49DCCFD96DC5DF56487436F5A1B18C4F5D34F65DDB48CB5EFFFFFFFFFFFFFFFF
Keys :16
Number of CPU thread: 4
BabyStep Thread 0: 0x1 -> 0x10000000
BabyStep Thread 1: 0x10000001 -> 0x20000000
BabyStep Thread 2: 0x20000001 -> 0x30000000
BabyStep Thread 3: 0x30000001 -> 0x40000000
[3.01 MKey/s][Cnt 2^28.52][01:57][3697.5MB]  Killed

**Comment by JeanLucPons** [comment]
Hi,
How many RAM do you have ?

**Comment by DvR4** [comment]
> chawla@i3-6100:~/jlp/BSGS$ ./bsgs in.txt BSGS v1.0 BabyStep:0x40000000 (2^30.00) Start:49DCCFD96DC5DF56487436F5A1B18C4F5D34F65DDB48CB5E0000000000000000 Stop :49DCCFD96DC5DF56487436F5A1B18C4F5D34F65DDB48CB5EFFFFFFFFFFFFFFFF Keys :16 Number of CPU thread: 4 BabyStep Thread 0: 0x1 -> 0x10000000 BabyStep Thread 1: 0x10000001 -> 0x20000000 BabyStep Thread 2: 0x20000001 -> 0x30000000 BabyStep Thread 3: 0x30000001 -> 0x40000000 [3.02 MKey/s][Cnt 2^28.48][01:55][3632.1MB] Killed chawla@i3-6100:~/jlp/BSGS$ ./bsgs in.txt
> BSGS v1.0
> BabyStep:0x40000000 (2^30.00)
> Start:49DCCFD96DC5DF56487436F5A1B18C4F5D34F65DDB48CB5E0000000000000000
> Stop :49DCCFD96DC5DF56487436F5A1B18C4F5D34F65DDB48CB5EFFFFFFFFFFFFFFFF
> Keys :16
> Number of CPU thread: 4
> BabyStep Thread 0: 0x1 -> 0x10000000
> BabyStep Thread 1: 0x10000001 -> 0x20000000
> BabyStep Thread 2: 0x20000001 -> 0x30000000
> BabyStep Thread 3: 0x30000001 -> 0x40000000
> [3.01 MKey/s][Cnt 2^28.52][01:57][3697.5MB] Killed

I think your RAM is full.

**Comment by DvR4** [comment]
> Hi,
> How many RAM do you have ?

Hi,JeanLuc
As I understand the number 40000000 is the Baby’s steps and takes up 9GB of RAM. If it is reduced to 4000000, will the calculation be correct?

**Comment by djarumlights** [comment]
Using 20000000 (2^29.00) at the top of in.txt takes up 4-5gb RAM.

**Comment by JeanLucPons** [comment]
Yes you can reduce the babystep number to save RAM, it affects the speed (2 times less RAM, 2 times slower) but calculation should be ok. If it is not, it is a bug.

**Comment by DvR4** [comment]
> Yes you can reduce the babystep number to save RAM, it affects the speed (2 times less RAM, 2 times slower) but calculation should be ok. If it is not, it is a bug.

Please tell me, is this tool applicable to bitcoin? If I set the key space of my choice, then I get an error
[108.08 MSort/s][Cnt 2^25.00][00s][1323.6MB]
Warning, range is not a multiple of nbThread
Warning, range is not a multiple of nbThread*1024

**Comment by JeanLucPons** [comment]
> Start:2000000000000000000000000000
> Stop :3FFFFFFFFFFFFFFFFFFFFFFFFFFF
I fixed the issue, the program was not working when number of step per thread > 2^64.
But 2^64 steps will take ages !
